connector/connect/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -360,13 +360,8 @@ class SparkConnectPlanner(session: SparkSession) {
     } else {
       logical.OneRowRelation()
     }
-    // TODO: support the target field for *.
     val projection =
-      if (rel.getExpressionsCount == 1 && rel.getExpressions(0).hasUnresolvedStar) {
-        Seq(UnresolvedStar(Option.empty))
-      } else {
-        rel.getExpressionsList.asScala.map(transformExpression).map(UnresolvedAlias(_))
-      }
+      rel.getExpressionsList.asScala.map(transformExpression).map(UnresolvedAlias(_))
```

Comment:
Though the code was written like this, I am seeing a problem that if the Expression is already a `Alias`, we should not wrap it into `UnresolvedAlias` again? 

I think you can write a test case like `SELECT 1 as col` then you will find the output schema is not `col` but a auto-generated alias.

Reply:
I am actually not sure if Catalyst will smartly see there is a wrapped `Alias` thus ignore the `UnresolvedAlias`. At least it is worth to verify this case.

Reply:
I think it will ignore. Can we add a test?

Reply:
+1 if there is no such test then add one.

Reply:
Thank you for your review @cloud-fan @amaliujia , I have added one test case for it, PTAL.